scheduler/pkg/kafka/config/tls.go

```diff
@@ -45,13 +47,15 @@ func AddKafkaSSLOptions(config kafka.ConfigMap) error {
 }
 
 func setupSASLSSLAuthentication(config kafka.ConfigMap) error {
-	cs, err := tls.NewCertificateStore(tls.ValidationOnly(true), tls.ValidationPrefix(EnvKafkaBrokerPrefix))
-	if err != nil {
-		return err
+	// Set the SASL mechanism
+	mechanism := tls.GetSASLMechanismFromEnv(tls.EnvSecurityPrefixKafka)
+	if (mechanism != tls.SASLMechanismPlain) && (mechanism != tls.SASLMechanismSCRAMSHA256) && (mechanism != tls.SASLMechanismSCRAMSHA512) {
```

Comment:
:thought_balloon: Related to my other comment, I'd probably have the validation in the env lookup so it can return an error appropriately.  That's then consistent for all users and handles the external/domain boundary at the earliest possible convenience